sos/plugins/__init__.py

```diff
@@ -806,13 +808,27 @@ def getmtime(path):
                     self._log_debug("skipping forbidden path '%s'" % _file)
                     continue
                 try:
-                    current_size += os.stat(_file)[stat.ST_SIZE]
+                    filestat = os.stat(_file)
                 except OSError:
                     self._log_info("failed to stat '%s'" % _file)
+                    continue
+                current_size += filestat[stat.ST_SIZE]
+                filetime = datetime.fromtimestamp(filestat[stat.ST_MTIME])
                 if sizelimit and current_size > sizelimit:
                     limit_reached = True
                     break
-                self._add_copy_paths([_file])
+
+                if _file.endswith('.gz') or re.search(r'-\d{8}$', _file):
+                    fileisarchive = True
+                else:
+                    fileisarchive = False
+
+                if (self.get_option('since') is None or
+                    fileisarchive is False or
+                    (fileisarchive is True and
+                     self.get_option('since') is not None and
+                     self.get_option('since') <= filetime)):
```

Comment:
This needs tidying up: if you're calling a method three times in one conditional, it's a sign it should have a variable in this scope. I think the logic would be more clear if you break it out into several lines: set variables, and then use those in the `if` statement if necessary. It's not obvious to a reader what behaviour you are aiming for here.

Reply:
Fixed in latest release